src/components/Posts/PostProvider.js

```diff
@@ -44,7 +44,7 @@ const createPost = (newPost) => new Promise((resolve, reject) => {
     },
     body: JSON.stringify(newPost)
```

Comment:
It's just few little changes for how your making the call.
```
fetch('http://localhost:8000/posts', {
    method: "POST",
    headers: {
      "Authorization": `Token ${localStorage.getItem("rare_token")}`
    },
    body: newPost
  })
```
Notice there's no content type and you don't need to stringify the newPost because you don't want the formdata to send as json. I'm a little surprised it didn't need `"Content-Type": "multipart/form-data"`but the mozilla docs don't have it on their example so it must be fine without it https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch#Uploading_a_file